rust/src/recalc/mod.rs

```diff
@@ -1,40 +1,43 @@
-use crate::FifoWindow;
-use alga::general::AbstractMonoid;
-use alga::general::Operator;
 use std::collections::VecDeque;
-use std::marker::PhantomData;
+
+use alga::general::AbstractMagma;
+use alga::general::Identity;
+
+use crate::FifoWindow;
+use crate::ops::AggregateOperator;
+use crate::ops::AggregateMonoid;
 
 #[derive(Clone)]
-pub struct ReCalc<Value, BinOp>
+pub struct ReCalc<BinOp>
 where
-    Value: AbstractMonoid<BinOp> + Clone,
-    BinOp: Operator,
+    BinOp: AggregateMonoid<BinOp> + AggregateOperator + Clone
 {
-    stack: VecDeque<Value>,
-    op: PhantomData<BinOp>,
+    stack: VecDeque<BinOp::Partial>,
```

Comment:
Can we rename this from stack to queue? That would be more intuitive, since we use it as a FIFO, not a LIFO.

Reply:
I think this should also be under a new issue and potentially a different PR. I'll create this one as well.